snow/validators/state.go

```diff
@@ -32,6 +32,15 @@ type State interface {
 		height uint64,
 		subnetID ids.ID,
 	) (map[ids.NodeID]*GetValidatorOutput, error)
+
+	// GetCurrentValidatorSet returns the current validators of the provided subnet
+	// and the current P-Chain height.
+	// Map is keyed by ValidationID.
+	// The returned map should not be modified.
```

Comment:
Why is this documented as not being able to be modified? I think everywhere this is implemented it can be modified.

Reply:
I think that's a left over from the previous attempt to use in-memory validators manager (from the `GetValidatorSet`). However wouldn't modifying the any validator in that map should still be illegal since we use currentStakers [here](https://github.com/ava-labs/avalanchego/blob/test-only-current-validators-api/vms/platformvm/state/state.go#L844-L853)?

Reply:
We create a new map every time, so it can always be owned by the caller. I'm not sure why it would matter where the data comes from that is put into the map

Reply:
removed it